lib/s3-s3.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * AWS.S3-like behavior with failover to a secondary S3 location
+ */
+
+'use strict';
+
+// Used in init() below, this function mimics AWS.S3
+var s3 = function (primaryS3, secondaryS3) {
```

Comment:
I will have actual examples in here at some point... but just to help understand this, the code is written so that you can keep your existing AWS.S3 calls pretty much the same.  But now instead of something like:
```
s3 = AWS.S3(awsConfig);
```
you instead do this:
```
s3 = s3s3.init(new AWS.S3(awsConfigPrimary), new AWS.S3(awsConfigSecondary));
```